test/suite/test_bug031.py

```diff
@@ -0,0 +1,146 @@
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # After eviction, we have:
+        # - Nothing in memory.
+        # - A record with a start ts 10 and stop ts 20 in the DS.
+        # - Nothing in the HS.
+
+        # Insert an update.
+        self.session.begin_transaction()
+        cursor[key] = value
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(30))
+
+        # Checkpoint.
+        self.session.checkpoint()
+
+        # After the checkpoint, we expect the following:
+        # - In memory, an update list with insert @ 30 (U3), remove @ 20 (U2) and insert @ 10 (U1).
+        # The two updates U1 and U2 should have the flag WT_UPDATE_PREPARE_RESTORED_FROM_DS as they
+        # were read back into memory from disk by the checkpoint.
+        # - The most recent insertion U3 in the DS.
+        # - A record with a start ts 10 and stop ts 20 in the HS.
+
+        # RTS.
+        self.conn.rollback_to_stable()
+
+        # Since the stable timestamp is 10, RTS marks all the updates performed at a time greater
+        # time than 10 as aborted. The update chain in memory is:
+        # U3 (aborted) -> U2 (aborted) -> U1.
+        # The DS now contains the insert @ 10 (U1) and the HS is empty.
+
+        # Insert but don't commit yet.
+        self.session.begin_transaction()
+        cursor[key] = value
+
+        # Evict everything.
+        evict_cursor = self.session.open_cursor(uri, None, 'debug=(release_evict)')
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # Since only the insert @ 10 (U1) is (re)written to disk, we now only have the two aborted
+        # updates in memory: U3 (aborted) -> U2 (aborted). The rest remains unchanged.
+
+        # Now commit a new insertion.
+        self.session.commit_transaction('commit_timestamp=' + self.timestamp_str(40))
+
+        # We now have an additional insert @ 40 (U4) in the update list in memory on top of the two
+        # aborted updates: U4 -> U3 (aborted) -> U2 (aborted)
+
+        # Evict everything.
+        evict_cursor = self.session.open_cursor(uri, None, 'debug=(release_evict)')
+        evict_cursor.set_key(key)
+        evict_cursor.search()
+        evict_cursor.reset()
+        evict_cursor.close()
+
+        # This moves the insertion @ 40 (U4) to the DS and the insertion @ 10 (U1) in the HS.
+    
+        # Read update at timestamp 10.
+        # The update chain is U4 -> U3 (aborted) -> U2 (aborted). Without the fix in WT-10522, we
+        # would exit early because of the flag WT_UPDATE_RESTORED_FROM_DS set on U2 and miss on
+        # appending U1.
+        self.session.begin_transaction('read_timestamp=' + self.timestamp_str(10))
+        cursor.set_key(key)
+        self.assertEquals(cursor.search(), 0)
```

Comment:
Doesn't the test need:
```
if __name__ == '__main__':
    wttest.run()
```

Reply:
Good question. Looks like it is something we haven't followed carefully if you do:
`grep -L "wttest.run()" test/suite/test_*`

Reply:
Added it to the test in 52584dc6b5a999602bc572f850859b2abba9fc3f.
Created WT-11079 to make it consistent.